cdk/domino_cdk/provisioners/eks/eks_nodegroup.py

```diff
@@ -277,16 +275,15 @@ def _handle_user_data(
 
         return mime_user_data
 
-    def _launch_template(self, scope, name: str, ng: config.eks.T_NodegroupBase, **opts) -> ec2.LaunchTemplate:
-        if ng.ami_id:
-            root_device_name = root_device_mapping(self.scope, ng.ami_id).name
-        else:
-            root_device_name = "/dev/xvda"  # This only works for AL2
-
-        opts = {
+    def _launch_template(self, scope, name: str, ng: config.eks.T_NodegroupBase, **kwargs) -> ec2.LaunchTemplate:
+        opts: Dict[Any, Any] = {
             "key_name": ng.key_name,
             "launch_template_name": f"{self.stack_name}-{name}",
-            "block_devices": [
+        }
+
+        if not ng.ami_id:
+            root_device_name = "/dev/xvda"  # This only works for AL2
```

Comment:
How they do remap if they use AL2 cloned image?

Reply:
Something like [packer](https://www.packer.io/docs/builders/amazon/ebs) can do this really easily.

Reply:
There is a neat AWS image builder as AWS online service. It allows to customize an existing AMI in AWS image repo by adding custom layers, like SSM. It doesn't run packer.

https://docs.aws.amazon.com/imagebuilder/latest/userguide/how-image-builder-works.html

Are we saying that if they build a customized AL2 image and need to more root device name to '/dev/xda' then they have to abandon AWS image building pipeline aaS and use offline 3rd party tool plus maintain their own building pipeline?

Reply:
After playing with EC3 image builder, 

on one hand, the disk size is an easy option when creating an image.

on the other hand, they will lose an ability to use the same image in different deployments by scaling the sizes from config.yaml.

on third hand, we do not give them ability to scale CPU/memory anyway.

Interesting, though, that LaunchTamplate API allows CPU selection https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.CfnLaunchTemplate.CpuOptionsProperty.html it doesn't allow memory selection.

Ideal solution, IMHO, is to 
1. Give them an option to scale CPU 
2. Give them a config option to explicitly indicate the device names and sizes. See, AL2 has more disks that /dev/xda https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html

If we cannot do this now, at least document that with custom AMI disk sizes are also not supported.

Reply:
Let me know what you think of 4676f075ecef58ab29a7f2c97d62e5f98df73294.